### ISSUE TITLE
How do manage API keys in aws-exports.js after publishing the mobile app?

### ISSUE THREAD

**Issue by V1j1** [aws-amplify/amplify-cli#2142]
** Which Category is your question related to? **
AWS App Sync
** What AWS Services are you utilizing? **
API based access to the back end
** Provide additional details e.g. code snippets **
I have an app that is working in my local. I now go to the AWS console and generate a new API key and paste it into the aws-exports.js file. I would like to know what's the right way to manage this once the app is published to the App Store.

Thanks in advance!

**Comment by kaustavghosh06** [comment]
@V1j1 Did you get a chance to take a look at our @auth docs which uses Cognito to manage your app users - https://aws-amplify.github.io/docs/cli-toolchain/graphql#auth?

**Comment by V1j1** [comment]
thanks @kaustavghosh06 ! That link is useful. However, I am trying to solve a different problem. I have a react-native app that uses aws-amplify and graphQL on the backend. I used the starter kit that came with the aws-amplify tutorial. I have completed building the application. I am using the AWS Cognito user pool to manage logins to my app. While creating the connectivity to my AWS account, I used AWS AppSync API key as the authentication mechanism and stored the API key that I got from the AWS Console (from the Settings option on AWS AppSync) in the aws-exports.js file. Now, I am realizing that this is not the right place to put my API key because the key needs to be rotated and I can't be releasing a new version everytime I update the key. What is the best alternate for this?
